MELA/fortran/mod_THiggs.F90

```diff
@@ -437,6 +465,149 @@ SUBROUTINE EvalAmp_QQB_TBARHB(MomExt,LO_Res_Unpol)
 END SUBROUTINE
 
 
+SUBROUTINE EvalAmp_GB_TWMH(MomExt,LO_Res_Unpol)
+use ModMisc
+use ModTTBHiggs
+use ModParameters
+use ModTopDecay
+
+implicit none
+real(8) :: MomExt(1:4,1:11),LO_Res_UnPol(-6:6,-6:6)
+complex(8) :: LOAmpab=0d0, LOAmpba=0d0
+
+complex(8) :: UBbD(4),UBbD2(4),Eg(4),Eg2(4),UBtD(4),ECw(4)
+integer, parameter :: iL=1,iR=2, H=3, t=4, w=5, bt=6,Wt=7,lept=8,nut=9, lepW=10,nuW=11
```

Comment:
What is the convention of ordering for lep/nu indices when either of the W charges are reversed?

Reply:
(or if one were to take W->qq' decay)

Reply:
due to charge conservation it is only possible to reverse both charges of both W bosons in t+H+W^- together resulting in the process tbar+H+W^+. The ordering is analogous to the assignment in ttbar (see mod_CrossSectionTH.F90 for particle IDs). 
In fact, all the routines in this function are valid for W->f'fbar (with f' and fbar being appropriate fermion/antifermion) and lep/nu are only placeholders.

Reply:
Sorry, then I don't understand the lep/nu vs sign convention. Which one of lep/nu indices are f and which ones are fbar? Are they reversed in t vs W decay by the opposite-charge pairing of Ws?

Reply:
for W^- the lep is the fermion and nu the antofermion.
for W^+ the opposite.

Reply:
and the assumption in the ME coding is that it is a t(+) W(-) final state?

Reply:
depends on the ME.
There are two:
EvalAmp_GB_TWMH for t(+) W(-) H

and 

EvalAmp_GBB_TBWPH for tbar(-) W(+) H

cf. existing single top s-/t-channel

Reply:
Understood. Is it possible to have one or two lines of comments above these two functions to note the difference and the lep/nu convention as well so that when they are interfaced, the momenta passed are ordered correctly?

Reply:
I will do that. 
I will get back to it later in the (my) evening.
Thanks so far.

Reply:
I have inserted respective comments